### DISCUSSION TITLE
Feature Idea: Skill And Potion Duration Bars - From Path of Exile AutoHotkey Macro

### DISCUSSION THREAD

**Discussion by leohero92** [Lailloken/Exile-UI#177] [Ideas & Feedback]
Hello Lailloken,

Could you please consider to add horizontal bar represented to it's duration when user press a key, something like this:
![Skill And Potion Duration Bars](https://user-images.githubusercontent.com/6971379/210063939-2c81e1fb-816c-4022-bc15-95219a9876a0.png)

This feature idea is based on a feature of Path of Exile AutoHotkey Macro
http://lutbot.com/#/

**Comment by Lailloken** [comment]
You can set up clone-frames for that. I think these are better for monitoring cooldowns/uptime because there are map-mods now that alter buff-duration and/or cooldown-recovery. But let me know if you have a use-case where clone-frames don't work.

![](https://user-images.githubusercontent.com/61888437/167854263-ce6c5da5-e5fa-4f4d-9ff9-f544859fa170.png)

**Reply by leohero92** [reply]
Hi @Lailloken, 

Thank you, 

I do know the clone-frame feature but I feel it's hard to track flask duration with these tiny bars, especially in PoE combat which is so chaotic and lack of visibility (as you know, a lot of on ground effect, after death effects, these projectiles all over screen.....). And you can see right in the posted picture, these **yellow** bars on **yellow** ground.

Same problem with skills, even if I set opacity as 5, it's still hard to track. If I have too much focus to track on these thing, I feel hard to focus in combat.

Clone-frame is still very useful.
But please consider to add these simple bars which can customize color, width, duration and position.

**Comment by Iphex** [comment]
Hi, I would like to jump onto this one for an interesting usecase. I am playing champion and have to trigger adrenaline every 20 seconds via gear swapping(midnight bargain 2x) (or, what others do is cf lifetap or something) and id LOVE a way to scan the buff bar for Adrenaline, show it besides my character and then even maybe play a sound if the buff disappeard. Same thing also works for molten shell, phase run etc. .

The easy way would be like the programm above, start an action on key press eg. I press x (gearswap) = trigger a 20 seconds bar that plays a sound (optional) at the end.

And this one is more farfetched, but scanning hp / es (mana) values and triggering an soundalarm / recording it for statitics upon reaching low values.

**Reply by Lailloken** [reply]
I'll think about it again. The reason why I haven't implemented something like this is that I think timers/bars by themselves look kind of cheap or cheesy. There's also the issue of buff expiration rate and temp chains desyncing the timings.

> and then even maybe play a sound if the buff disappeard

> scanning hp / es (mana) values and triggering an soundalarm

That sounds tryhard and like something GGG wouldn't allow.
